test/models/submission/analysis_test.rb

```diff
@@ -72,10 +72,89 @@ class Submission::AnalysisTest < ActiveSupport::TestCase
     assert_equal expected, analysis.feedback_html
   end
 
-  test "has_feedback?" do
-    refute create(:submission_analysis, data: { comments: nil }).has_feedback?
-    refute create(:submission_analysis, data: { comments: [] }).has_feedback?
-    assert create(:submission_analysis, data: { comments: ['foobar'] }).has_feedback?
+  test "has_comments?" do
+    refute create(:submission_analysis, data: { comments: nil }).has_comments?
+    refute create(:submission_analysis, data: { comments: [] }).has_comments?
+    assert create(:submission_analysis, data: { comments: ['foobar'] }).has_comments?
+  end
+
+  test "informative comments: none" do
+    comments = ["ruby.two-fer.incorrect_default_param"]
+    analysis = create :submission_analysis, data: { comments: comments }
+
+    assert_equal 0, analysis.num_informative_comments
+    refute analysis.has_informative_comments?
+  end
+
+  test "informative comments: one" do
+    analysis = create :submission_analysis, data: { comments: [{
+      "comment" => "ruby.two-fer.string_interpolation",
+      "type": "informative"
+    }] }
+
+    assert_equal 1, analysis.num_informative_comments
+    assert analysis.has_informative_comments?
```

Comment:
Would it make be useful to verify that the informative comment contains the correct values (as in check for the actual comment identifier)?

Reply:
There's quite a lot of tests for that further up in the file. I don't see us actually ever having a method `analysis.informative_comments` which contains those comments, so I'd be inventing that method purely for testing purposes. However, if we do end up with that, then I'll definitely test for the content, yeah.